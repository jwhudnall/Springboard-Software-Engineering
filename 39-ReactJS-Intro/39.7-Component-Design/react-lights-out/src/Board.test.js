import { render, screen, fireEvent } from "@testing-library/react";
import Board from "./Board";

// Smoke test
it("should render", () => {
  render(<Board />);
});

// Snapshot test
it("should match snapshot (all lit)", () => {
  // 100% chance of lit to maintain snapshot consistency
  const { asFragment } = render(<Board chanceLightStartsOn={1} />);
  expect(asFragment()).toMatchSnapshot();
});

it("should match snapshot (none lit)", () => {
  // 100% chance of lit to maintain snapshot consistency
  const { asFragment } = render(<Board chanceLightStartsOn={0} />);
  // screen.debug();
  expect(asFragment()).toMatchSnapshot();
});

it("should display winning message (none lit)", () => {
  render(<Board chanceLightStartsOn={0} />);
  const msg = screen.getByText("You Win!");
  expect(msg).toBeInTheDocument();
});

it("should toggle all surrounding cells upon click", () => {
  render(<Board chanceLightStartsOn={1} />);
  const cellToBeClicked = screen.getByTestId("1-2");
  fireEvent.click(cellToBeClicked);
  const above = screen.getByTestId("0-2");
  const right = screen.getByTestId("1-3");
  const below = screen.getByTestId("2-2");
  const left = screen.getByTestId("1-1");
  expect(cellToBeClicked).not.toHaveClass("Cell-lit");
  expect(above).not.toHaveClass("Cell-lit");
  expect(right).not.toHaveClass("Cell-lit");
  expect(left).not.toHaveClass("Cell-lit");
  expect(below).not.toHaveClass("Cell-lit");
});
