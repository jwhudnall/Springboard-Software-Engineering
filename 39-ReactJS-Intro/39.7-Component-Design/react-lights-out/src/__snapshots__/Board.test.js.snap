// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot (all lit) 1`] = `
<DocumentFragment>
  <table>
    <tbody>
      <tr>
        <td
          class="Cell Cell-lit"
          data-testid="0-0"
        />
        <td
          class="Cell Cell-lit"
          data-testid="0-1"
        />
        <td
          class="Cell Cell-lit"
          data-testid="0-2"
        />
        <td
          class="Cell Cell-lit"
          data-testid="0-3"
        />
        <td
          class="Cell Cell-lit"
          data-testid="0-4"
        />
      </tr>
      <tr>
        <td
          class="Cell Cell-lit"
          data-testid="1-0"
        />
        <td
          class="Cell Cell-lit"
          data-testid="1-1"
        />
        <td
          class="Cell Cell-lit"
          data-testid="1-2"
        />
        <td
          class="Cell Cell-lit"
          data-testid="1-3"
        />
        <td
          class="Cell Cell-lit"
          data-testid="1-4"
        />
      </tr>
      <tr>
        <td
          class="Cell Cell-lit"
          data-testid="2-0"
        />
        <td
          class="Cell Cell-lit"
          data-testid="2-1"
        />
        <td
          class="Cell Cell-lit"
          data-testid="2-2"
        />
        <td
          class="Cell Cell-lit"
          data-testid="2-3"
        />
        <td
          class="Cell Cell-lit"
          data-testid="2-4"
        />
      </tr>
      <tr>
        <td
          class="Cell Cell-lit"
          data-testid="3-0"
        />
        <td
          class="Cell Cell-lit"
          data-testid="3-1"
        />
        <td
          class="Cell Cell-lit"
          data-testid="3-2"
        />
        <td
          class="Cell Cell-lit"
          data-testid="3-3"
        />
        <td
          class="Cell Cell-lit"
          data-testid="3-4"
        />
      </tr>
      <tr>
        <td
          class="Cell Cell-lit"
          data-testid="4-0"
        />
        <td
          class="Cell Cell-lit"
          data-testid="4-1"
        />
        <td
          class="Cell Cell-lit"
          data-testid="4-2"
        />
        <td
          class="Cell Cell-lit"
          data-testid="4-3"
        />
        <td
          class="Cell Cell-lit"
          data-testid="4-4"
        />
      </tr>
    </tbody>
  </table>
</DocumentFragment>
`;

exports[`should match snapshot (none lit) 1`] = `
<DocumentFragment>
  <h1>
    You Win!
  </h1>
</DocumentFragment>
`;
